src/viam/sdk/robot/client.cpp

```diff
@@ -57,6 +47,147 @@ using viam::robot::v1::Status;
+    if (proto.has_kinematics()) {
+        fsconfig.kinematics = struct_to_map(proto.kinematics());
+    }
+    return fsconfig;
+}
+
+bool operator==(const RobotClient::frame_system_config& lhs,
+                const RobotClient::frame_system_config& rhs) {
+    return lhs.frame == rhs.frame && map_to_struct(lhs.kinematics).SerializeAsString() ==
+                                         map_to_struct(rhs.kinematics).SerializeAsString();
+}
+
+viam::robot::v1::Status RobotClient::status::to_proto() const {
+    viam::robot::v1::Status proto;
+    if (name) {
+        *proto.mutable_name() = name->to_proto();
+    }
+    if (status_map) {
+        *proto.mutable_status() = map_to_struct(status_map);
+    }
+    if (last_reconfigured) {
+        *proto.mutable_last_reconfigured() = time_pt_to_timestamp(*last_reconfigured);
+    }
+    return proto;
+}
+
+RobotClient::status RobotClient::status::from_proto(const viam::robot::v1::Status& proto) {
+    status status;
+    if (proto.has_name()) {
+        status.name = Name::from_proto(proto.name());
+    }
+    if (proto.has_status()) {
+        status.status_map = struct_to_map(proto.status());
+    }
+    if (proto.has_last_reconfigured()) {
+        status.last_reconfigured = timestamp_to_time_pt(proto.last_reconfigured());
+    }
+    return status;
+}
+
+bool operator==(const RobotClient::status& lhs, const RobotClient::status& rhs) {
+    return lhs.name == rhs.name &&
+           map_to_struct(lhs.status_map).SerializeAsString() ==
+               map_to_struct(rhs.status_map).SerializeAsString() &&
+           lhs.last_reconfigured == rhs.last_reconfigured;
+}
+
+viam::robot::v1::Operation RobotClient::operation::to_proto() const {
+    viam::robot::v1::Operation proto;
+    *proto.mutable_id() = id;
+    *proto.mutable_method() = method;
+    if (session_id) {
+        *proto.mutable_session_id() = *session_id;
+    }
+    if (arguments) {
+        *proto.mutable_arguments() = map_to_struct(arguments);
+    }
+    if (started) {
+        *proto.mutable_started() = time_pt_to_timestamp(*started);
+    }
+    return proto;
+}
+
+RobotClient::operation RobotClient::operation::from_proto(const viam::robot::v1::Operation& proto) {
+    operation op;
+    op.id = proto.id();
+    op.method = proto.method();
+    if (proto.has_session_id()) {
+        op.session_id = proto.session_id();
+    }
+    if (proto.has_arguments()) {
+        op.arguments = struct_to_map(proto.arguments());
+    }
+    if (proto.has_started()) {
+        op.started = timestamp_to_time_pt(proto.started());
+    }
+    return op;
+}
+
+bool operator==(const RobotClient::operation& lhs, const RobotClient::operation& rhs) {
+    return lhs.id == rhs.id && lhs.method == rhs.method && lhs.session_id == rhs.session_id &&
+           lhs.arguments == rhs.arguments && lhs.started == rhs.started;
+}
+
+struct RobotClient::Impl {
+    Impl(std::unique_ptr<RobotService::Stub> stub) : stub_(std::move(stub)) {}
```

Comment:
Yes, after making it so that `stub_` is no longer a pointer, it then gives me an error like so:
```
note: in instantiation of function template specialization 'std::make_unique<viam::sdk::RobotClient::impl, std::unique_ptr<viam::robot::v1::RobotService::Stub>>' requested here
      impl_(std::make_unique<impl>(RobotService::NewStub(channel_))) {}
                 ^
```
called when we try to call: 
```
RobotClient::RobotClient(std::shared_ptr<ViamChannel> channel)
    : viam_channel_(channel),
      channel_(channel->channel()),
      should_close_channel_(false),
      impl_(std::make_unique<impl>(RobotService::NewStub(channel_))) {}
```